src/asciidocParser.ts

```diff
@@ -2,42 +2,37 @@ import * as vscode from 'vscode'
 import * as path from 'path'
 import { spawn } from 'child_process'
 import { AsciidoctorWebViewConverter } from './asciidoctorWebViewConverter'
+import { Asciidoctor } from "@asciidoctor/core";
+
+const asciidoctorFindIncludeProcessor = require('./asciidoctorFindIncludeProcessor')
 
 const asciidoctor = require('@asciidoctor/core')
 const docbook = require('@asciidoctor/docbook-converter')
 const kroki = require('asciidoctor-kroki')
 const processor = asciidoctor()
 const highlightjsBuiltInSyntaxHighlighter = processor.SyntaxHighlighter.for('highlight.js')
 const highlightjsAdapter = require('./highlightjs-adapter')
-processor.ConverterFactory.register(new AsciidoctorWebViewConverter(), ['html5'])
 
 export class AsciidocParser {
-  public html: string = ''
-  public document = null
-  public processor = null
   private stylesdir: string
+  public baseDocumentIncludeItems = null
 
-  constructor (private readonly filename: string, private errorCollection: vscode.DiagnosticCollection = null) {
-    const extensionContext = vscode.extensions.getExtension('asciidoctor.asciidoctor-vscode')
-    this.stylesdir = vscode.Uri.joinPath(extensionContext.extensionUri, 'media').toString()
-  }
-
-  public getAttribute (name: string) {
-    return (this.document == null) ? null : this.document.getAttribute(name)
+  constructor (extensionUri: vscode.Uri, private errorCollection: vscode.DiagnosticCollection = null) {
```

Comment:
This class desperately needs to be rewritten.
We should have dissociate the `load` function from the `convert` function and avoid using mutable class variables.

The `load` and `convert` functions should be pure.

Reply:
I'm happy to look at this, perhaps we can discuss on the issue.